src/main/java/xyz/nucleoid/extras/lobby/item/TaterGuidebookItem.java

```diff
@@ -0,0 +1,231 @@
+import net.minecraft.util.Hand;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.TypedActionResult;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraft.world.chunk.Chunk;
+import xyz.nucleoid.extras.lobby.block.TinyPotatoBlock;
+import xyz.nucleoid.extras.mixin.lobby.ThreadedAnvilChunkStorageAccessor;
+
+public class TaterGuidebookItem extends Item implements PolymerItem {
+    private static final Text MISSING_SYMBOL = Text.literal("❌").formatted(Formatting.RED);
+    private static final Text FOUND_SYMBOL = Text.literal("✔").formatted(Formatting.GREEN);
+    private static final Text TOO_MANY_SYMBOL = Text.literal("✔").setStyle(Style.EMPTY.withColor(0x055005));
+
+    private static final int RECORD_COOLDOWN = 2 * SharedConstants.TICKS_PER_SECOND;
+
+    private static final String TATER_POSITIONS_KEY = "tater_positions";
+
+    public TaterGuidebookItem(Settings settings) {
+        super(settings);
+    }
+
+    @Override
+    public TypedActionResult<ItemStack> use(World world, PlayerEntity user, Hand hand) {
+        var stack = user.getStackInHand(hand);
+
+        if (!world.isClient() && user.isCreativeLevelTwoOp()) {
+            var player = (ServerPlayerEntity) user;
+            var taterPositionMap = getTaterPositions(stack);
+
+            if (user.isSneaking()) {
+                recordToGuidebook(player, taterPositionMap, stack);
+            } else {
+                showGuidebook(player, taterPositionMap, stack);
+            }
+
+            return TypedActionResult.success(stack, world.isClient());
+        }
+
+        return TypedActionResult.pass(stack);
+    }
+
+    @Override
+    public Item getPolymerItem(ItemStack stack, ServerPlayerEntity player) {
+        return Items.WRITTEN_BOOK;
+    }
+
+    private static void recordToGuidebook(ServerPlayerEntity player, SetMultimap<Item, BlockPos> taterPositions, ItemStack stack) {
+        int initialCount = taterPositions.size();
+
+        var pos = new BlockPos.Mutable();
+        var chunkManager = player.getWorld().getChunkManager();
+
+        var chunkStorage = chunkManager.threadedAnvilChunkStorage;
+        var accessor = (ThreadedAnvilChunkStorageAccessor) (Object) chunkStorage;
+
+        for (var holder : accessor.callEntryIterator()) {
+            var chunk = holder.getWorldChunk();
+
+            if (chunk != null) {
+                recordChunk(chunk, pos, taterPositions);
+            }
+        }
+
+        setTaterPositions(stack, taterPositions);
+
+        player.getItemCooldownManager().set(stack.getItem(), RECORD_COOLDOWN);
+
+        int difference = taterPositions.size() - initialCount;
+        player.sendMessage(Text.translatable("text.nucleoid_extras.tater_guidebook.recorded", difference), true);
+    }
+
+    private static void recordChunk(Chunk chunk, BlockPos.Mutable pos, SetMultimap<Item, BlockPos> taterPositions) {
+        for (int z = 0; z < 16; z++) {
```

Comment:
`Chunk.forEachBlockMatchingPredicate`? Should be a lot more efficient, as it checks palette before doing a linear scan 🙂